File: mainActivity/src/main/java/com/sierrawireless/avphone/auth/Authentication.java
package com.sierrawireless.avphone.auth;

import java.util.Date;

public class Authentication {
    private String accessToken;
    private Date expirationDate;
    
    public String getAccessToken() {
        return accessToken;
    }
    
    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }
    
    public boolean isExpired(Date date) {
        return date.after(this.expirationDate);
    }
    
    public boolean isExpired() {
        return isExpired(new Date());
    }
    
    public Date getExpirationDate() {
        return this.expirationDate;
    }
    
    public void setExpirationDate(Date date) {
        this.expirationDate = date;
    }
}

File: mainActivity/src/main/java/com/sierrawireless/avphone/auth/Authentication.kt
package com.sierrawireless.avphone.auth

import java.util.Date

class Authentication {
    var accessToken: String? = null
    var expirationDate: Date? = null

    val isExpired: Boolean
        get() = isExpired(Date())

    fun isExpired(date: Date): Boolean {
        return date.after(this.expirationDate)
    }
}
